is-state/is-state.typ

```diff
@@ -0,0 +1,227 @@
+
+Ein eigener Fork hat auch den Vorteil gegenüber einem Branch, dass das Hosting einfacher zu bewerkstelligen ist.
+
+== Architektur
+
+=== Daten
+
+Alle Daten zu #gls("module", display: "Modulen"), #gls("category", display: "Kategorien"), #gls("focus", display: "Spezialisierung") und #gls("std", display: "Studienordnungen"), welche der Planer nutzt, sind öffentlich #link("https://studien.ost.ch") zugänglich.
+
+Diese Daten werden von einem Python Crawler gesammelt, verarbeitet und als JSON Dateien im Data Repository abgelegt.
+
+Die Detailseiten der Studienordnungen dienen hier als Einstiegspunkt. Informationen zu den geltenden Kategorien und benötigten #glspl("ects"), die möglichen Spezialisierungen und die zugehörigen Module können so entommen werden.
+
+Die Aktualisierung der Daten erfolgt manuel. Vor Beginn eines jeden Semester führt ein Maintainer den Crawler lokal aus, überprüft die Änderungen der Daten auf Spezialfälle und erstellt anschliessend ein neues Tag für die Daten. Zuletzt kann die verwendete Version der Daten im Client über den Tag in der URL angepasst werden. 
+
+Folgend sind die relevanten Felder für eine Studienordnung gelistet, mit Beispielsdaten anhand #link("https://studien.ost.ch/allStudies/10191_I.json").
+```json
+{
+  "kredits": [
+    {
+      "kategorien": [
+        {
+          "bezeichnung": "Aufbau",
+          "kuerzel": "I_Auf"
+        },
+        ...
+      ],
+      "minKredits": 48
+    }
+  ],
+  "zuordnungen": [
+    {
+      "bezeichnung": "Application Architecture",
+      "kuerzel": "M_AppArch",
+      "url": "allModules/28236_M_AppArch.json",
+      "istAbschlussArbeit": false,
+      "istPflichtmodul": false,
+      "semEmpfehlung": 7,
+      "kategorien": [
+        {
+          "bezeichnung": "Aufbau",
+          "kuerzel": "I_Auf",
+          "kreditpunkte": 4
+        },
+        ...
+      ]
+    }
+  ],
+  "spezialisierungen": [
+    {
+      "bezeichnung": "Frontend Engineering",
+      "kuerzel": "FrEng",
+      "url": "allStudies/10193_FrEng.json"
+    },
+    ...
+  ]
+}
+```
+Für jedes Modul wird anhand der `"zuordnungen[].url"` eine Anfrage gemacht, woraus das folgende Feld relevant ist.
+```json
+{
+  "kreditpunkte": 4
+}
+```
+Für jede Spezialisierung wird anhand der `"spezialisierungen[].url"` eine Anfrage gemacht, woraus die folgenden Felder relevant sind.
+```json
+{
+  "zuordnungen": [
+    {
+      "kuerzel": "M_AppArch"
+    },
+    ...
+  ]
+}
+```
+
+Das Diagram @crawler stellt den Ablauf, bei dem der Crawler die oben erwähnten Daten verarbeitet, grob dar.
+#figure(image("Crawler.drawio.png"), caption: [Ablauf des Crawlers]) <crawler>
+
+Wie bereits erwähnt, wird das Format der Daten für die weitere Nutzung angepasst.
```

Comment:
Ich sehe nicht, wo das vorher bereits erwähnt wurde.
```suggestion
Das Datenformat wird für die weitere Nutzung angepasst.
```

Reply:
Hier könntest du noch erwähnen, weshalb das Format angepasst wird. Z.B. irgendwie so "Um sicherzustellen, dass die Daten von der Applikation effizient abgefragt und verarbeitet werden können, wird ihr Format entsprechend angepasst. Auf diese Weise werden zudem redundante Daten für die Applikation nicht übernommen." 
Wollen wir in der Arbeit immer mit "Planer" das Produkt referenzieren oder machen wir Applikation draus?

Reply:
S isch sozsege im Diagram vom Crawler erwähnt wurde.
Aber fair, d Erklärig wär guet.

Mer chend luege, ob "Planer" oder "Applikation". Für mich wär au beides ok. Dass "Applikation" eigentlich nume de Client meint und "Planer" isch s ganze. Aber evtl wärs professioneller, nume "Applikation" zbruche... Mer chenne ja mol GPT froge.

Reply:
Mache mer "Semesterplaner", wenns us "User-Sicht" isch und "Applikation", wenn mer s Ganze us technischer Sicht meine.